src/test/java/no/unit/nva/cristin/projects/FetchOneCristinProjectTest.java

```diff
@@ -0,0 +1,139 @@
+import java.net.HttpURLConnection;
+import java.util.Map;
+import no.unit.nva.cristin.projects.model.nva.NvaProject;
+import no.unit.nva.testutils.HandlerRequestBuilder;
+import nva.commons.apigateway.GatewayResponse;
+import nva.commons.core.Environment;
+import nva.commons.core.ioutils.IoUtils;
+import org.apache.commons.codec.Charsets;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class FetchOneCristinProjectTest {
+
+    private static final String EMPTY_JSON = "{}";
+    private static final String CRISTIN_GET_PROJECT_ID_NOT_FOUND_RESPONSE_JSON =
+        "cristinGetProjectIdNotFoundResponse.json";
+    private static final String API_RESPONSE_ONE_PROJECT_JSON =
+        "api_response_one_cristin_project_to_nva_project.json";
+    private static final String CRISTIN_PROJECT_WITHOUT_INSTITUTION_AND_PARTICIPANTS_JSON =
+        "cristinProjectWithoutInstitutionAndParticipants.json";
+    private static final String API_RESPONSE_GET_PROJECT_WITH_MISSING_FIELDS_JSON =
+        "api_response_get_project_with_missing_fields.json";
+    private static final String NOT_AN_ID = "Not an ID";
+    private static final String DEFAULT_ID = "9999";
+
+    private CristinApiClient cristinApiClientStub;
+    private Environment environment = new Environment();
+    private Context context;
+    private ByteArrayOutputStream output;
+    private FetchOneCristinProject handler;
+
+    @BeforeEach
+    void setUp() {
+        cristinApiClientStub = new CristinApiClientStub();
+        context = mock(Context.class);
+        output = new ByteArrayOutputStream();
+        handler = new FetchOneCristinProject(cristinApiClientStub, environment);
+    }
+
+    @Test
+    void handlerReturnsEmptyJsonWhenIdIsNotFound() throws Exception {
+        cristinApiClientStub = spy(cristinApiClientStub);
+        doReturn(getReader(CRISTIN_GET_PROJECT_ID_NOT_FOUND_RESPONSE_JSON))
+            .when(cristinApiClientStub).fetchGetResult(any());
+
+        handler = new FetchOneCristinProject(cristinApiClientStub, environment);
+        GatewayResponse<NvaProject> response = sendQueryWithId(DEFAULT_ID);
+
+        assertEquals(OBJECT_MAPPER.readTree(EMPTY_JSON), OBJECT_MAPPER.readTree(response.getBody()));
+    }
+
+    @Test
+    void handlerReturnsBadRequestWhenIdIsNotANumber() throws Exception {
+        GatewayResponse<NvaProject> response = sendQueryWithId(NOT_AN_ID);
+        assertEquals(HttpURLConnection.HTTP_BAD_REQUEST, response.getStatusCode());
+    }
+
+    @Test
+    void handlerReturnsNvaProjectFromTransformedCristinProjectWhenIdIsFound() throws Exception {
+        GatewayResponse<NvaProject> response = sendQueryWithId(DEFAULT_ID);
+        var expected = getReader(API_RESPONSE_ONE_PROJECT_JSON);
+        assertEquals(OBJECT_MAPPER.readTree(expected), OBJECT_MAPPER.readTree(response.getBody()));
+    }
+
+    @Test
+    void handlerReturnsEmptyJsonWhenFetchFromBackendFails() throws Exception {
```

Comment:
@brinxmat  Should we maybe return BadGateway here?

Reply:
If it fails, it is an error